### DISCUSSION TITLE
Add row number in table view

### DISCUSSION THREAD

**Discussion by Dominic-Preap** [directus/directus#23053] [Feature Requests]
### Summary

It would be nice to have auto sequence row number display at the start column on table view in collections.

![image](https://github.com/user-attachments/assets/5aac808a-6576-4e5d-9edd-0cff0451eecf)

### Basic Example

Just like material-react-table

https://www.material-react-table.com/docs/guides/row-numbers

### Motivation

For distinguish record and for follow up records in which page without looking and pagination. Better for people who get used to excel spreadsheet format row number.

### Detailed Design

The implementation would be straightforward just page * total-size and +1 for each row but where to put the configure at. I think we have we option model editor with checkbox to show row display number.  

### Requirements List

N/A

### Drawbacks

None, just for aesthetics look and uniqueness.

### Alternatives

N/A

### Adoption Strategy

N/A

**Comment by rijkvanzanten** [comment]
I'm wondering if that would be confusing with the primary key, which is the actual row's identifier number 🤔

**Reply by Dominic-Preap** [reply]
@rijkvanzanten sometimes the requirements doesn't want to show the id or the id might get deleted then data is in the wrong sequence or some primary implement using uuid.

But most of the time, we don't want to show the id just title or name and some related fields

Like in the mantine example we can use header name hashtag # to show a separation between id and row number. Most of the end user does not have much tech knowledge, so using primary key as row number still might confuse if some record is deleted.

**Reply by rijkvanzanten** [reply]
Good point!
